### ISSUE TITLE
ScanNet++V2 Benchmarking Discrepancy

### ISSUE THREAD

**Issue by Nik-V9** [ByteDance-Seed/Depth-Anything-3#29]
Hey folks, congrats on the amazing release!

I wanted to follow up regarding the benchmarking trends you showed for MapAnything on ScanNet++ in your paper. The bad performance of MapAnything on DTU checks out with my internal testing due to the released MapAnything not being trained on object centric data and it's metric training priors not aligning well with the toy scale objects presented in the DTU dataset. 

However, the ScanNet++ and other indoor dataset trends don't seem to align with my intuition. I wonder if this difference stems from how your ScanNet++ dataset is processed? Do you folks use the iphone RGB images and poses (I believe these are lower quality in terms of precision)?

<img width="500" height="200" alt="Image" src="https://github.com/user-attachments/assets/aa7630e8-68ec-4641-9eb8-3c2313aa7efa" />

<img width="500" height="200" alt="Image" src="https://github.com/user-attachments/assets/5ba2b75e-1485-4c9e-9089-474bf6dbcc36" />

**I've added Depth Anything 3 to the MapAnything benchmarking framework and I see very impressive results on ETH3D and subpar performance on ScanNet++V2 and TartanAirV2-WB:**

**Average:**
<img width="4767" height="2435" alt="Image" src="https://github.com/user-attachments/assets/ae44b058-f624-4577-b692-fdac80b14921" />

**ETH3D:**
<img width="4767" height="2435" alt="Image" src="https://github.com/user-attachments/assets/b5c58fd7-ba01-4985-bb47-dac53392a70e" />

**ScanNet++V2:**
<img width="4767" height="2435" alt="Image" src="https://github.com/user-attachments/assets/693cf338-bb54-4567-b1b5-809886beb2a1" />

**TartanAirV2:**
<img width="4767" height="2435" alt="Image" src="https://github.com/user-attachments/assets/7985285d-a9d6-45db-8406-463bd207a330" />

Also I visualized the outputs (on datasets and in-the-wild image sets) and think that there is no bug with the benchmarking/inference. The code for your reference (I'm testing with 504 longest side and dinov2 image normalization):
```
# Copyright (c) Meta Platforms, Inc. and affiliates.

# This source code is licensed under the Apache License, Version 2.0
# found in the LICENSE file in the root directory of this source tree.

"""
Inference wrapper for Depth Anything 3
"""

import numpy as np
import torch

from depth_anything_3.api import DepthAnything3

from mapanything.models.external.vggt.utils.geometry import closed_form_inverse_se3
from mapanything.models.external.vggt.utils.rotation import mat_to_quat
from mapanything.utils.geometry import (
    convert_ray_dirs_depth_along_ray_pose_trans_quats_to_pointmap,
    convert_z_depth_to_depth_along_ray,
    depthmap_to_camera_frame,
    get_rays_in_camera_frame,
)


class DA3Wrapper(torch.nn.Module):
    def __init__(
        self,
        name,
        torch_hub_force_reload,
        hf_model_name,
    ):
        super().__init__()
        self.name = name
        self.torch_hub_force_reload = torch_hub_force_reload
        self.hf_model_name = hf_model_name

        # Load pre-trained weights
        if not torch_hub_force_reload:
            # Initialize the DA3 model from huggingface hub cache
            print("Loading DA3 from huggingface cache ...")
            self.model = DepthAnything3.from_pretrained(
                self.hf_model_name,
            )
        else:
            # Initialize the DA3 model
            self.model = DepthAnything3.from_pretrained(self.hf_model_name, force_download=True)
        
        # Get the dtype for DA3 inference
        # bfloat16 is supported on Ampere GPUs (Compute Capability 8.0+)
        self.dtype = (
            torch.bfloat16
            if torch.cuda.get_device_capability()[0] >= 8
            else torch.float16
        )

    def forward(self, views):
        """
        Forward pass wrapper for DA3

        Assumption:
        - All the input views have the same image shape.

        Args:
            views (List[dict]): List of dictionaries containing the input views' images and instance information.
                                Each dictionary should contain the following keys:
                                    "img" (tensor): Image tensor of shape (B, C, H, W).
                                    "data_norm_type" (list): ["dinov2"]

        Returns:
            List[dict]: A list containing the final outputs for all N views.
        """
        # Get input shape of the images, number of views, and batch size per view
        batch_size_per_view, _, height, width = views[0]["img"].shape
        num_views = len(views)

        # Check the data norm type
        data_norm_type = views[0]["data_norm_type"][0]
        assert data_norm_type == "dinov2", (
            "DA3 expects DINOv2 normalization for the input images"
        )

        # Concatenate the images to create a single (B, V, C, H, W) tensor
        img_list = [view["img"] for view in views]
        images = torch.stack(img_list, dim=1)

        # Run the DA3 model
        with torch.autocast("cuda", dtype=self.dtype):
            results = self.model(images, export_feat_layers=[])

        # Need high precision for transformations
        with torch.autocast("cuda", enabled=False):
            res = []
            for view_idx in range(num_views):
                # Get the extrinsics, intrinsics, depth map for the current view
                curr_view_extrinsic = results["extrinsics"][:, view_idx, ...]
                curr_view_extrinsic = closed_form_inverse_se3(
                    curr_view_extrinsic
                )  # Convert to cam2world
                curr_view_intrinsic = results["intrinsics"][:, view_idx, ...]
                curr_view_depth_z = results["depth"][:, view_idx, ...]
                curr_view_depth_z = curr_view_depth_z.squeeze(-1)
                curr_view_confidence = results["depth_conf"][:, view_idx, ...]

                # Get the camera frame pointmaps
                curr_view_pts3d_cam, _ = depthmap_to_camera_frame(
                    curr_view_depth_z, curr_view_intrinsic
                )

                # Convert the extrinsics to quaternions and translations
                curr_view_cam_translations = curr_view_extrinsic[..., :3, 3]
                curr_view_cam_quats = mat_to_quat(curr_view_extrinsic[..., :3, :3])

                # Convert the z depth to depth along ray
                curr_view_depth_along_ray = convert_z_depth_to_depth_along_ray(
                    curr_view_depth_z, curr_view_intrinsic
                )
                curr_view_depth_along_ray = curr_view_depth_along_ray.unsqueeze(-1)

                # Get the ray directions on the unit sphere in the camera frame
                _, curr_view_ray_dirs = get_rays_in_camera_frame(
                    curr_view_intrinsic, height, width, normalize_to_unit_sphere=True
                )

                # Get the pointmaps
                curr_view_pts3d = (
                    convert_ray_dirs_depth_along_ray_pose_trans_quats_to_pointmap(
                        curr_view_ray_dirs,
                        curr_view_depth_along_ray,
                        curr_view_cam_translations,
                        curr_view_cam_quats,
                    )
                )

                # Append the outputs to the result list
                res.append(
                    {
                        "pts3d": curr_view_pts3d,
                        "pts3d_cam": curr_view_pts3d_cam,
                        "ray_directions": curr_view_ray_dirs,
                        "depth_along_ray": curr_view_depth_along_ray,
                        "cam_trans": curr_view_cam_translations,
                        "cam_quats": curr_view_cam_quats,
                        "conf": curr_view_confidence,
                    }
                )

        return res
```

**Comment by tobiasfshr** [comment]
+1 on this, in my tests DA3 1.4B performs only slightly better than VGGT on ScanNet++

**Comment by haotongl** [comment]
@Nik-V9 Hi Nikhil,

Thanks again for your attention and testing.

1. We found that the default poses in ScanNet++ are often inaccurate (e.g., 5f99900f09). When running NeRF or Gaussian Splatting without camera pose optimization, many scenes produce strange or unstable results. We suspect this is mainly due to motion blur and textureless frames from the iPhone captures.

2. To address this, we re-ran COLMAP for all ScanNet++ scenes with the following changes:
    a. We re-extracted frames to remove blurry images.
    b. We jointly calibrated a fisheye camera, which provides a wider field of view and avoids blurry frames. Joint calibration also leads to more accurate pose calibration.
    c. We used COLMAP’s exhaustive matcher and mapper. Although the COLMAP mapper can take several days per scene, we found it to be very necessary for obtaining reliable camera poses (overall, recalibrating poses for all scenes may take hundreds of GPU nodes running for several days).
    d. We have released the processed scenes here, so please feel free to download and test them:
https://huggingface.co/datasets/haotongl/scannetpp_zipnerf/tree/main

3. To reproduce the results in our paper, we recommend evaluating the following scenes. For each scene, please:
    a. Use 100 iPhone images. (You should randomly select 100 images for each scene).
    b. Use our recalibrated poses for pose evaluation, and
    c. Use mesh_aligned_0.05.ply (scanned by the accurate FARO sensor) as the reference for reconstruction evaluation.

The exact scenes of our evaluation are as follows:
[
    "09c1414f1b",
    "1ada7a0617",
    "40aec5fffa",
    "3e8bba0176",
    "acd95847c5",
    "578511c8a9",
    "5f99900f09",
    "c4c04e6d6c",
    "f3d64c30f8",
    "7bc286c1b6",
    "c5439f4607",
    "286b55a2bf",
    "fb5a96b1a2",
    "7831862f02",
    "38d58a7a31",
    "bde1e479ad",
    "9071e139d9",
    "21d970d8de",
    "bcd2436daf",
    "cc5237fd77",
]

**Comment by haotongl** [comment]
We are preparing to release the full evaluation code as well. However, it will take about two weeks since I’ve been quite busy recently. Please stay tuned.

**Comment by Nik-V9** [comment]
Hey Haotong, thanks for the quick clarification!

Any reason why you folks preferred to reprocess/recalibrate the iphone images as opposed to using the DSLR images which have a very high quality registration to the FARO scan? I guess it's always hard to quantify the accuracy of COLMAP labels for benchmarking?

Also, just a note in case you are interested, we provide that process to get posed RGB-D for the DSLR here: https://github.com/facebookresearch/map-anything/tree/main/data_processing
